// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package autofill;

option optimize_for = LITE_RUNTIME;

message StateEntry {
  // Full name representing the state entry unique within a country.
  // Example:
  //  "California" for "CA", "California", "The Golden State".
  //  "Bavaria" for "BY", "Bavaria", "Bayern".
  optional string canonical_name = 1;

  // Abbreviations corresponding to the state entry.
  repeated string abbreviations = 2;

  // Alternative names of the state.
  repeated string alternative_names = 3;
}

message StatesInCountry {
  // Two digit country code.
  optional string country_code = 1;

  // All the states belonging to the country.
  repeated StateEntry states = 2;
}
